### ISSUE TITLE
Setter with no parameter is valid and produces wrong code

### ISSUE THREAD

**Issue by enjoydambience** [square/kotlinpoet#952]
A setter with no parameter just generates the word "set" verbatim; with no body, and no newline.

A setter without a parameter should probably just throw a exception.
```kotlin
FileSpec.builder("","Property")
    .addType(TypeSpec.classBuilder("Foo")
        .addProperty(PropertySpec.builder("prop", String::class)
            .setter(FunSpec.setterBuilder()
                .addStatement("println(%S)","I have a body!")
                .build())
            .build())
        .build())
    .build()
```
produces:
```
import kotlin.String

class Foo {
  val prop: String
    set}
```

**Comment by Egorand** [comment]
Kotlin allows parameterless setters, e.g.:
```kotlin
var foo: String = ""
  private set
```
That said, the fact that the body is simply ignored if the parameter is not provided is a bug. A PR would be appreciated!

**Comment by Egorand** [comment]
Fixed by #954
